python/ray/util/client/api.py

```diff
@@ -312,27 +312,57 @@ def _internal_kv_initialized(self) -> bool:
         # initialized yet.
         return True
 
-    def _internal_kv_exists(self, key: bytes) -> bool:
+    def _internal_kv_exists(
+        self, key: Union[str, bytes], *, namespace: Optional[Union[str, bytes]] = None
+    ) -> bool:
         """Hook for internal_kv._internal_kv_exists."""
-        return self.worker.internal_kv_exists(_as_bytes(key))
+        return self.worker.internal_kv_exists(
+            _as_bytes(key), namespace=_as_bytes(namespace)
+        )
 
-    def _internal_kv_get(self, key: bytes) -> bytes:
+    def _internal_kv_get(
+        self, key: Union[str, bytes], *, namespace: Optional[Union[str, bytes]] = None
+    ) -> bytes:
         """Hook for internal_kv._internal_kv_get."""
-        return self.worker.internal_kv_get(_as_bytes(key))
+        return self.worker.internal_kv_get(
+            _as_bytes(key), namespace=_as_bytes(namespace)
+        )
 
     def _internal_kv_put(
-        self, key: bytes, value: bytes, overwrite: bool = False
+        self,
+        key: Union[str, bytes],
+        value: Union[str, bytes],
+        overwrite: bool = True,
+        *,
+        namespace: Optional[Union[str, bytes]] = None,
     ) -> bool:
         """Hook for internal_kv._internal_kv_put."""
-        return self.worker.internal_kv_put(_as_bytes(key), _as_bytes(value), overwrite)
+        return self.worker.internal_kv_put(
+            _as_bytes(key), _as_bytes(value), overwrite, namespace=_as_bytes(namespace)
+        )
 
-    def _internal_kv_del(self, key: bytes) -> None:
+    def _internal_kv_del(
+        self,
+        key: Union[str, bytes],
+        *,
+        del_by_prefix: bool = False,
+        namespace: Optional[Union[str, bytes]] = None,
+    ) -> int:
         """Hook for internal_kv._internal_kv_del."""
-        return self.worker.internal_kv_del(_as_bytes(key))
+        return self.worker.internal_kv_del(
+            _as_bytes(key), del_by_prefix=del_by_prefix, namespace=_as_bytes(namespace)
```

Comment:
is `del_by_prefix ` tested with the client mode?

Reply:
Yes, see `test_basic_5.py`